x-pack/plugins/code/common/git_url_utils.ts

```diff
@@ -6,28 +6,27 @@
 
 import GitUrlParse from 'git-url-parse';
 
-export function isValidGitUrl(
+// return true if the git url is valid, otherwise throw Error with
+// exact reasons.
+export function validateGitUrl(
   url: string,
   hostWhitelist?: string[],
   protocolWhitelist?: string[]
 ): boolean {
-  try {
-    const repo = GitUrlParse(url);
+  const repo = GitUrlParse(url);
 
-    let isHostValid = true;
-    if (hostWhitelist && hostWhitelist.length > 0) {
-      const hostSet = new Set(hostWhitelist);
-      isHostValid = hostSet.has(repo.source);
+  if (hostWhitelist && hostWhitelist.length > 0) {
+    const hostSet = new Set(hostWhitelist);
+    if (!hostSet.has(repo.source)) {
+      throw new Error('Git url host is not whitelisted.');
     }
+  }
 
-    let isProtocolValid = true;
-    if (protocolWhitelist && protocolWhitelist.length > 0) {
-      const protocolSet = new Set(protocolWhitelist);
-      isProtocolValid = protocolSet.has(repo.protocol);
+  if (protocolWhitelist && protocolWhitelist.length > 0) {
+    const protocolSet = new Set(protocolWhitelist);
+    if (!protocolSet.has(repo.protocol)) {
```

Comment:
if no protocol specified, we should say protocol is needed for url

Reply:
this has nothing to do with missing protocol. here it's checking if the given protocol is in the whitelist.